### PR TITLE
docs: update API STYLE.md

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

Commit Message: Add some examples and adjust the development flow to make it clear.
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/STYLE.md [hunk 1]
```diff
@@ -148,19 +153,14 @@ To add an extension config to the API, the steps below should be followed:
...
-1. Make sure your proto imports the v3 extension config proto (`import "udpa/annotations/status.proto";`)
-1. Make sure your proto is tracked as ready to be used
-   (`option (udpa.annotations.file_status).package_version_status = ACTIVE;`).
-   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD) under `active_protos`. Without it `proto_format.sh` will delete the proto file.
-1. Add a reference to the v3 extension config in (2) in [api/BUILD](BUILD) under `v3_protos`. Without this, `proto_format.sh` will erase any imports of your proto file from other proto files.
-1. If you introduce a new extension category, you'll also need to add its name
-   under `categories` in: [tools/extensions/extensions_schema.yaml](../tools/extensions/extensions_schema.yaml).
 1. Run `./tools/proto_format/proto_format.sh fix`. Before running the script, you will need to commit your local changes to make them effective,
    otherwise it will skip proto_format.sh due to no API change detected. This should regenerate the `BUILD` file and reformat `foobar.proto` as needed.
```
Comment (phlax) [inline api/STYLE.md:165]:
```suggestion
1. Run `./tools/proto_format/proto_format.sh fix`. **Before running the script**, you will need to **commit your local changes**. By adding the commit, the tool will recognize the change, and will regenerate the `BUILD` file and reformat `foobar.proto` as needed. If you have not followed any of the above steps correctly `proto_format.sh` may remove some of the files that you added. If that is the case you can revert to the committed state, and try again once any issues are resolved.
```

**Review by phlax — COMMENTED** [review]
@daixiang0 this looks good - thanks for this

my suggestion would be not to add the "...otherwise this will happen lines" re proto_format.sh as i think it makes the points less clear

rather i have suggested editing the `proto_format.sh` line itself with a bit more information about potentially messing your filesystem

File: api/STYLE.md [hunk 2]
```diff
@@ -129,6 +126,14 @@ To add an extension config to the API, the steps below should be followed:
        deps = ["@com_github_cncf_udpa//udpa/annotations:pkg"],
    )
    ```
+1. If this is still WiP and subject to breaking changes, please tag it
+   `option (xds.annotations.v3.file_status).work_in_progress = true;` and
+   optionally hide it from the docs (`[#not-implemented-hide:]`).
+1. Make sure your proto imports the v3 extension config proto (`import "udpa/annotations/status.proto";`)
+1. Make sure your proto is tracked as ready to be used
+   (`option (udpa.annotations.file_status).package_version_status = ACTIVE;`).
+   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD) under `active_protos`. Without it `proto_format.sh` will delete the proto file.
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply api/STYLE.md:135]: Updated.

File: api/STYLE.md [hunk 3]
```diff
@@ -129,6 +126,14 @@ To add an extension config to the API, the steps below should be followed:
...
+1. Add a reference to the v3 extension config in (2) in [api/BUILD](BUILD) under `v3_protos`. Without this, `proto_format.sh` will erase any imports of your proto file from other proto files.
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply api/STYLE.md:136]: Done.

File: api/STYLE.md (hunk 1 above)
Comment (daixiang0) [reply api/STYLE.md:165]: Done.

**Comment by daixiang0** [comment]
@phlax thanks for your review, I have updated all.

File: api/STYLE.md [hunk 4]
```diff
@@ -129,7 +126,15 @@ To add an extension config to the API, the steps below should be followed:
...
    ```
-1. Update [source/extensions/extensions_metadata.yaml](../source/extensions/extensions_metadata.yaml)
+1. If this is still WiP and subject to breaking changes, please tag it
+   `option (xds.annotations.v3.file_status).work_in_progress = true;` and
+   optionally hide it from the docs (`[#not-implemented-hide:]`).
+1. Make sure your proto imports the v3 extension config proto (`import "udpa/annotations/status.proto";`)
+1. Make sure your proto is tracked as ready to be used
+   (`option (udpa.annotations.file_status).package_version_status = ACTIVE;`).
+   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD) under `active_protos`.
+1. Add a reference to the v3 extension config in (2) in [api/BUILD](BUILD) under `v3_protos`.
```
Comment (phlax) [inline api/STYLE.md:136]:
```suggestion
1. Add a reference to the v3 extension config in [api/BUILD](BUILD) under `v3_protos`.
```

**Review by phlax — CHANGES_REQUESTED** [review]
@daixiang0 one small nit and i think should be good to land

File: api/STYLE.md (hunk 4 above)
Comment (daixiang0) [reply api/STYLE.md]: Done.

**Comment by daixiang0** [comment]
@phlax please review again.

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Pull request merged**